foresight/environment/environment_info_support.py

```diff
@@ -0,0 +1,83 @@
+import os, logging
+from foresight.environment.git.git_helper import GitHelper
+from foresight.environment.git.git_env_info_provider import GitEnvironmentInfoProvider
+from foresight.environment.github.github_environment_info_provider import GithubEnvironmentInfoProvider
+from foresight.environment.gitlab.gitlab_environment_info_provider import GitlabEnvironmentInfoProvider
+from foresight.environment.jenkins.jenkins_environment_info_provider import JenkinsEnvironmentInfoProvider
+from foresight.environment.travisci.travisci_environment_info_provider import TravisCIEnvironmentInfoProvider
+from foresight.environment.circleci.circleci_environment_info_provider import CircleCIEnvironmentInfoProvider
+from foresight.environment.bitbucket.bitbucket_environment_info_provider import BitbucketEnvironmentInfoProvider
+from foresight.test_runner_tags import TestRunnerTags
+
+LOGGER = logging.getLogger(__name__)
+
+class EnvironmentSupport:
+
+    ENVIRONMENTS_VARS = {
+        "GITHUB_REPOSITORY": GithubEnvironmentInfoProvider,
+        "GITLAB_CI": GitlabEnvironmentInfoProvider,
+        "JENKINS_HOME": JenkinsEnvironmentInfoProvider,
+        "JENKINS_URL": JenkinsEnvironmentInfoProvider,
+        "TRAVIS": TravisCIEnvironmentInfoProvider,
+        "CIRCLECI": CircleCIEnvironmentInfoProvider,
+        "BITBUCKET_GIT_HTTP_ORIGIN": BitbucketEnvironmentInfoProvider,
+        "BITBUCKET_GIT_SSH_ORIGIN": BitbucketEnvironmentInfoProvider
+    }
+    environment_info = None
+
+
+    @classmethod
+    def init(cls):
+        """First check git provider, then iterate over ENVIRONMENTS_VARS dict.
+        """
+        try:
+            LOGGER.debug("Checking ci environments...")
+            for key, clz in cls.ENVIRONMENTS_VARS.items():
+                LOGGER.debug("Current key, clz: {}, {}".format(key,clz))
+                if os.getenv(key):
```

Comment:
who set this env var?

Reply:
These are the default environment variables set by appropriate CI Tools. I also add links near them as comment.